Service/DRCService.php

```diff
@@ -361,23 +381,48 @@ public function inhoudHandler(array $data, array $configuration): array
                 $file->setSize(0);
             }
 
-            if ($data['inhoud'] !== null && filter_var($data['inhoud'], FILTER_VALIDATE_URL) === false) {
+            if ($data['inhoud'] !== null && $data['inhoud'] !== '' && filter_var($data['inhoud'], FILTER_VALIDATE_URL) === false) {
                 $file->setSize(mb_strlen(base64_decode($data['inhoud'])));
                 $file->setBase64($data['inhoud']);
-            } else if(($data['inhoud'] === null && $data['link'] === null) || $file->getBase64() === '' && $data['bestandsomvang'] !== null) {
-                $parts    = ceil($data['bestandsomvang'] / 1000000);
+            } else if ((($data['inhoud'] === null || filter_var($data['inhoud'], FILTER_VALIDATE_URL) === $data['inhoud']) && ($data['link'] === null || $data['link'] === '')) && isset($this->data['body']['bestandsomvang']) === true) {
```

Comment:
This is getting a little hard to read, not sure what to do about it though. Maybe move some of this to a variable instead 🤔